contracts/samples/TokenPaymaster.sol

```diff
@@ -142,8 +145,12 @@ contract TokenPaymaster is BasePaymaster, UniswapHelper, OracleHelper {
             }
             uint256 tokenAmount = weiToToken(preChargeNative, cachedPriceWithMarkup);
             SafeERC20.safeTransferFrom(token, userOp.sender, address(this), tokenAmount);
-            context = abi.encodePacked(tokenAmount, userOp.maxFeePerGas, userOp.maxPriorityFeePerGas, userOp.sender);
-            validationResult = 0;
+            context = abi.encode(tokenAmount, userOp.maxFeePerGas, userOp.maxPriorityFeePerGas, userOp.sender);
+            validationResult = _packValidationData(
+                false,
+                uint48(cachedPriceTimestamp + tokenPaymasterConfig.priceMaxAge),
```

Comment:
storage optimization: instead referencing 2 storage slots on each call, use 
`cachedPriceExpirationTimestamp = timestamp + priceMaxAge`

Reply:
Please elaborate where does the `timestamp` variable come from?